service/controller/resource/app/desired.go

```diff
@@ -272,3 +313,34 @@ func newUserConfig(cr apiv1alpha2.Cluster, appSpec key.AppSpec, configMaps map[s
 
 	return userConfig
 }
+
+func (r *Resource) getCatalogIndex(ctx context.Context, catalogName string) ([]byte, error) {
+	client := &http.Client{}
+
+	var err error
+	var catalog *g8sv1alpha1.AppCatalog
+	{
+		catalog, err = r.g8sClient.ApplicationV1alpha1().AppCatalogs().Get(ctx, catalogName, metav1.GetOptions{})
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	url := strings.TrimRight(catalog.Spec.Storage.URL, "/") + "/index.yaml"
+	request, err := http.NewRequestWithContext(ctx, http.MethodGet, url, &bytes.Buffer{}) // nolint: gosec
```

Comment:
I'm assuming this is an external call? If so we might need to have a certain backoff retry mechanism in case we do not get it the first time, wdyt?

Reply:
Good point!